File: frontend/src/types/policy.ts
export interface BuyerPolicyEnvelope {
  max_budget: number;
  max_delivery_days: number;
  min_sla_percent: number;
}

export interface VendorPolicyEnvelope {
  min_price: number;
  min_delivery_days: number;
  max_sla_percent: number;
}

export interface NegotiationPolicies {
  buyer: BuyerPolicyEnvelope;
  vendor: VendorPolicyEnvelope;
}

export const BUYER_CONSTRAINTS = {
  max_budget: { min: 1, max: 50000 },
  max_delivery_days: { min: 1, max: 45 },
  min_sla_percent: { min: 2.0, max: 100 },
} as const;

export const VENDOR_CONSTRAINTS = {
  min_price: { min: 42000, max: 1000000 },
  min_delivery_days: { min: 30, max: 365 },
  max_sla_percent: { min: 0.1, max: 5.0 },
} as const;

File: backend/app/schemas/policy.py
from pydantic import BaseModel, Field


class BuyerPolicyEnvelope(BaseModel):
    max_budget: float = Field(..., le=50000, gt=0, description="Maximum budget ceiling")
    max_delivery_days: int = Field(..., le=45, gt=0, description="Latest acceptable delivery")
    min_sla_percent: float = Field(..., ge=2.0, le=100.0, description="Lowest acceptable SLA penalty")


class VendorPolicyEnvelope(BaseModel):
    min_price: float = Field(..., ge=42000, description="Minimum price floor")
    min_delivery_days: int = Field(..., ge=30, description="Fastest possible delivery")
    max_sla_percent: float = Field(..., le=5.0, gt=0, description="Maximum acceptable SLA penalty")


class NegotiationPolicies(BaseModel):
    buyer: BuyerPolicyEnvelope
    vendor: VendorPolicyEnvelope
